server/src/main/java/com/google/musicanalysis/site/MusixParamBuilder.java

```diff
@@ -0,0 +1,77 @@
+package com.google.musicanalysis.site;
+
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+
+/*
+    Builds parameter strings based off what the user wants.
+    This is structured the way it is becuase I will be adding more
+    to each function. I will clean it up later as well.
+*/
+public class MusixParamBuilder {
+
+  private String intendedUse;
+
+  // All possible parameters I can receive
+  private String trackId;
+  private String trackName;
+  private String albumId;
+  private String artistName;
+
+  /**
+   * Constructor.
+   *
+   * @param intendedUse determines what action we will send to the API
+   * @param params array of parameters which get assigned to different values depending on
+   *     intendedUse
+   */
+  public MusixParamBuilder(String intendedUse, String... params) {
+    this.intendedUse = intendedUse;
+
+    switch (intendedUse) {
+      case "track.search":
+        this.trackName = params[0];
+        this.artistName = params[1];
+        break;
+      case "track.lyrics.get":
+        this.trackId = params[0];
+        break;
+      default:
+        throw new IllegalArgumentException("arguments required");
+    }
+  }
+
+  /** Filters out the correct parameter string and returns it */
+  public String grabFilteredParams() {
```

Comment:
nit: name this getFilteredParams.

Reply:
+1, "get" is the preferred verbiage, consistent with the concept of a "getter"